src/aktualizr_primary/main.cc

```diff
@@ -147,10 +147,10 @@ int main(int argc, char *argv[]) {
     } else {
       aktualizr.RunForever().get();
     }
-    r = 0;
+    r = EXIT_SUCCESS;
   } catch (const std::exception &ex) {
     LOG_ERROR << ex.what();
-    r = -1;
+    r = EXIT_FAILURE;
   }
   conn.disconnect();
```

Comment:
This may throw, I think. Should we move it under the try block?

Reply:
I guess, you mean line #155 (disconnect) ? If it's moved under the try block then the disconnect() won't be called if one of the previous statements throws exception.

Reply:
Yes, this line. I know, it's not the scope of this PR, but since we already discussing it, maybe change it to `scoped_connection` then?

Reply:
NP, IMHO, scoped_connection may it little bit better but it does not solve the original problem - i.e. proper tear-down/exit if disconnect() throws.

Reply:
Why? If we change it to scoped_connection and move into the try block, then it will automatically disconnect if an exception is thrown in this block.

Reply:
'This(disconnect) may throw' was the original issue statement and it still the issue if it's called in scope of exception handling flow.

Reply:
Ah, sorry, I got your previous comment wrong. With `scoped_connection` there is no `disconnect` call, it will be disconnected on the object destruction and destructors don't throw.
Also, I don't think it's a problem that `disconnect` may throw (which I'm not 100% sure, btw), but that it will be an uncaught exception and we don't print anything.

Reply:
~scoped_connection()
      {
        disconnect();
      }

If disconnect() throws then scoped_connection's ctor throws too.

Reply:
I'm pretty sure that boost follows the guideline that destructor cannot throw, which leads us to the conclusion that disconnect doesn't throw ;) But it would be nicer if they used `noexcept`.